src/data/proxy/DDBPreferenceCorrelationGraph.java

```diff
@@ -108,4 +115,18 @@ public Preference getPreference(String id, PreferenceCategory category) {
             return new DDBPreferenceAdapter(item).toObject();
         }
     }
+    
+    private List<Preference> getPreferences(Page<Item, ScanOutcome> page) {
+    	Iterator<Item> iterator = page.iterator();
+    	return Stream.generate(iterator::next).map(item -> new DDBPreferenceAdapter(item).toObject()).collect(Collectors.toList());
+    }
+
+	@Override
+	public Stream<List<Preference>> batchGetPreferences(
```

Comment:
Why does this need to return a Stream? It seems like it would be more logical for it to just return a Collection of Preferences.